gateway/webrtc/client.js

```diff
@@ -410,66 +409,72 @@ function handleOffer(offer, sender) {
+                        createAndSendAnswer(offer,sender);
                     } else {
-                        callBtn.style.display='inline';
-                        callToUsernameInput.style.display='inline';
-                        overlay.style.visibility='hidden';
-                        hangUpBtn.style.display='none';
-                        
+                        showOffCall();
                         leave();
                     }
         }
         });
     } else {
-        internalCreateAnswer(offer,sender);
+       createAndSendAnswer(offer,sender);
     }
     console.log("Exiting handleOffer");
 };
 
-function internalCreateAnswer(offer, sender ) {
 
-    yourConn.setRemoteDescription(new RTCSessionDescription(offer)).catch(function (e) { console.log("Remote error", e) ;});
+function createAndSendAnswer(offer, sender ) {
 
     //create an answer to an offer
-    yourConn.createAnswer().then(function(answer) {
+    yourConn.createAnswer()
+    .then(function(answer) {
         console.log("Entering createAnswer signalling", answer);
-        
-        callBtn.style.display='none';
-        callToUsernameInput.style.display='none';
-        hangUpBtn.style.display='inline';
-        overlay.style.visibility='visible';
-        overlay.innerText='Connected to '+connectedUser;
-        yourConn.setLocalDescription(answer);
-
-        send({
-            type: "answer",
-            answer: answer
+        showInCall();
+        yourConn.setLocalDescription(new RTCSessionDescription(answer))
+        .then(function () {
+            var sdp = reconstructSDP (answer.sdp);
+            answer.sdp = sdp;
+            send({
+                type: "answer",
+                answer: answer
+            });
+        })
+        .catch(function(error) {
+            console.log("Error when setting local description", error);
         });
         console.log("Exiting createAnswer signalling");
+    })
 
-    }).catch(function(error) {
+    .catch(function(error) {
         console.log("Error when creating an answer", error);
     });
+
 }
-var answerReceived = false;
+
 //when we got an answer from a remote user
 function handleAnswer(answer) {
     console.log("Entering handleAnswwer");
-    yourConn.setRemoteDescription(new RTCSessionDescription(answer)).catch(function (e) { console.log("Remote error", e); });
+    yourConn.setRemoteDescription(answer)
+    .catch(function (e) { 
+        console.log("Error when setting remote description", e); 
+    });
+
     answerReceived = true;
     console.log("Exiting handleAnswer");
 };
 
+
 //when we got an ice candidate from a remote user
 function handleCandidate(candidate) {
     console.log("Entering handleCandidate", candidate);
-    yourConn.addIceCandidate(new RTCIceCandidate(candidate)).catch(e=>{
-	      console.log("Error: Failure during addIceCandidate()", e);
-    });;
+    if (candidate.candidate.indexOf("host") < 0) {
```

Comment:
There is a <0 there instead of >0. So it means "add all candidates OTHER than host"